packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -641,6 +663,28 @@ function propagateParentContextChanges(
   workInProgress.flags |= DidPropagateContext;
 }
 
+function checkIfComparedContextValuesChanged(
+  oldComparedValue: mixed,
+  newComparedValue: mixed,
+): boolean {
+  if (isArray(oldComparedValue) && isArray(newComparedValue)) {
+    if (oldComparedValue.length !== newComparedValue.length) {
+      return true;
+    }
+
+    for (let i = 0; i < oldComparedValue.length; i++) {
+      if (!is(newComparedValue[i], oldComparedValue[i])) {
+        return true;
+      }
+    }
+  } else {
+    if (!is(newComparedValue, oldComparedValue)) {
+      return true;
+    }
+  }
```

Comment:
Not really sure it's worth leaving this in b/c you can't really just change the compiler without also changing this comparison logic since you'd end up with incidental index based comparisons if you ever didn't return a wrapping array. Since its de facto part of the API might as well make anything that violates this API error so you know you messed something up in the compiler

Reply:
Removed the `is` check here in favor of leaning into the array return contract. Also typed the inputs as Arrays and check against `lastSelectedValue` being available before entering comparison function